### DISCUSSION TITLE
Synthmorph - Parameters selection

### DISCUSSION THREAD

**Discussion by EvanBeal** [voxelmorph/voxelmorph#372] [Q&A] [answered]
Hello ! 

I contacted you a few weeks ago because I find the Synthmorph approach very interesting and I wanted to try to train a model following your methodology. Thanks to your answer and your advice in the tutorial you provided, I was able to do it and I find it very promising. 

However, I was not able to get as good results as one can get using the `sm_shapes` model that is available on your GitHub and that was used in your article. Therefore, I was wondering if I was setting the different parameters used in the generation of the label maps and grayscale images correctly. 

For the drawing of the image (shape generation) with the function `ne.utils.augment.draw_perlin`, I am using `scales=(16, 32, 64)` and a `max_std=1` and for the warp, `scales=(8, 16, 32)` and `max_std=3`. 
This is done to generate `100 label maps` with a `size of 160x160x192` and `26 labels`. 
For the rest of the process, I am using the default parameters that are present in the file `scripts/tf/train_synthmorph.py`, except for `vel-std` that I have set to 3 instead of 0.5. 
I am aware that increasing the number of filters to 256 instead of 64 would improve the registration model for example, however I feel like this is probably not the only difference between the model I am training and the one that you provided on GitHub as there are many parameters involved. 

So, I would be interested to know if it is possible to tell me how you were able to get the `sm_shapes` model (what are the different parameters you used for the label maps generation) ? And what is different from what I am currently doing ? 

Thank you for taking the time to respond to me, I really look forward to continuing to discover all the possibilities and benefits that this method can bring.

**Comment by mu40** [comment, marked as answer]
Hi @EvanBeal, It’s great to hear that you found the tutorial helpful for setting up your own training runs!

It sounds like you are using the recommended parameters. The warp strength `--vel-std` is an important setting, and we sometimes found a value of 2 to work better than 3 for some datasets, so this would be something to try.

I would definitely encourage you to train larger models and to train them for several thousand "epochs". While this does take a long time, the 256-filter models performed substantially better in our experiments than networks with 64 filters, likely because more capacity is needed for tackling the wide variety of synthetic training shapes.

Also be sure to use the `--same-subj` flag, as it is not guaranteed that a plausible transform exists between two independently sampled shape label maps.

**Reply by EvanBeal** [reply]
Thanks @mu40 for the quick and detailed response!

I'm glad to hear that I'm close to the optimal parameters for label map generation. 

I will definitely try to play with the `--vel-std` parameter and see how it affects my results and regarding the model architecture, I will follow your advice to try to improve the model performance.

Yes, thank you for the reminder of the `--same-subj` flag, I used it for the different training runs but it is useful to check it indeed. 

So, thank you very much for these explanations and tips, I will now try these recommendations and see if it works well on my own training runs!
